File: #0867.transpose-matrix.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {

        int row = matrix.size();
        int col = matrix[0].size();
        vector<int> tmp;
        vector<vector<int>> result;
        for (int i = 0; i < col; i++) {
            tmp.clear();
            for (int j = 0; j < row; j++) {
                tmp.push_back(matrix[j][i]);
            }
            result.push_back(tmp);
        }
        return result;
    }
};

File: #0867.transpose-matrix.js
/**
 * @param {number[][]} matrix
 * @return {number[][]}
 */
var transpose = function(matrix) {
    const result = Array.from(
        { length: matrix[0].length },
        () => new Array(matrix.length).fill(0)
    );

    for (let i = 0 ; i < matrix.length ; i++) {
        for (let j = 0; j < matrix[0].length; j++) {
            result[j][i] = matrix[i][j];
        }
    }
    return result;
};
